src/charm.py

```diff
@@ -198,6 +212,26 @@ def _get_hydra_endpoint_info(self) -> Optional[str]:
 
         return oauth2_provider_url
 
+    def _get_login_ui_endpoint_info(self, key: str) -> Optional[str]:
+        if self.model.relations[self._login_ui_relation_name]:
+            try:
+                login_ui_endpoints = self.login_ui_endpoints.get_login_ui_endpoints()
+                login_ui_url = login_ui_endpoints[key]
+                return login_ui_url
+            except LoginUIEndpointsRelationDataMissingError:
+                logger.info("No login ui endpoint-info relation data found")
+                if key == "default_url":
```

Comment:
Well, we want to use the constant DEFAULT_BROWSER_RETURN_URL, if there's no relation between kratos-operator and an instance of login-ui operator, or there's a problem with accessing login-ui endpoints (for example there's no public facing proxy). I get that it looks bloated the way it is right now, but checking these conditions are necessary, because we only want to use the default if something goes wrong. Taking key=="default_url" to the root of the method would then replicate the nested condition checking.

With that being said, I can change it, but returning the default every time the key is "default_url" is not an option.

Reply:
I would suggest to handle this in the jinja template with something like:
```
         {{ default_browser_return_url | d("http://example-default-return-url.com") }}
```

Reply:
changed charm.py and jinja template. default browser is now handled by the template. I used an if else statement, because an logical or printed "None" value.